### DISCUSSION TITLE
Does this make the smartShunt redundant?

### DISCUSSION THREAD

**Discussion by CRaccani** [Louisvdw/dbus-serialbattery#110] [Q&A] [answered]
Hi, thank you for your hard work in this software.

My question is: Does the JKbms now act as a shunt/battery monitor so as to make the need for the Victron smartShunt unnecessary?

I have been using a smartshunt in my Victron ESS system together with a JK BMS with my DIY lithium battery.
The smartShunt has been seen by the Victron system as a battery monitor and so controls the SoC logic.
The JK BMS is not known to the system of course.
The smartShunt and the JK BMS both count coulombs in and out- thus keeping track of the remaining capacity.
Both the smartShunt and the JK BMS monitor the battery state but the problem is that they are seldom in sync for any length of time.
Should I then not use the smartShunt if the JK BMS becomes the battery monitor for the system and rely on the accuracy of the JK BMS alone.?

Thank you.

**Comment by Louisvdw** [comment, marked as answer]
Hi
The driver will enable your BMS to be the battery monitor. So yes you do not need the SmartShunt as this will be handled by the BMS. 
The smartshunt is a great piece of hardware and very accurate, but is does not have any internal knowledge of your battery and this is where the BMS shines. 
So yes I dod prefr to rather use the BMS as it is much safer.
But these Chinese BMS do not all come with the perfect calibration of of the factory. You can caibrate your BMS's shunt which is what is counting the columns, but I cant remember seeing such a function on the JKBMS app.
Any imballances on any of the cells will also influence the SOC calculation. In this case the Smurtshunt would not know about it as it does not have cell level data, so then the BMS has the better results.

It is always better to have one source of the truth. And for that I would suggest the BMS with cell level is the better source. Even better would be if you have calebrated.

**Reply by CRaccani** [reply]
Can't ask for a better answer than that. Thank you and for being so quick to answer.
Regarding calibration: In the App select Settings/Basic Settings, Using a multimeter you can enter a voltage and current as calibration values.

Going to install your driver tonight.

Thanks again.

**Reply by CRaccani** [reply]
OK, success, works just fine.
Had to make a cable as Victron connector not same as BMS, but once done works just fine.

Am on Venus OS 2.66 by the way. Have been working just fine on this version for my use. Are there any problems/restrictions when upgrading to latest version.
See attached photos for, pictures say a thousand words, extra info.
Thanks again
![IMG_0080](https://user-images.githubusercontent.com/64036712/158698338-f2f0285c-5ff3-4595-a4be-b147ed5a59f7.JPG)
![IMG_0081](https://user-images.githubusercontent.com/64036712/158698358-94e5a2d7-7240-49a6-abe1-96c3eb8accb5.JPG)
![IMG_0082](https://user-images.githubusercontent.com/64036712/158698386-6de85ccd-b922-4417-9657-3a570a6b6af3.JPG)
![IMG_0084](https://user-images.githubusercontent.com/64036712/158698401-b46cb6f4-0946-4482-a6af-e4b1c589d19f.PNG)
.

**Reply by Louisvdw** [reply]
Perfect. Looks good. 
Up to VenusOS V2.73 the driver works the same as on V2.66
From V2.80+ the driver still works, but there are a few niggles still to be sorted out. If you only have 1 BMS connected it will be fine. If you also have ET112 or similar power meter there are some issues after a power failure that the meter is picked up as a battery. Also if you have 2 or more BMS connected it does not work in V2.80+

**Reply by thorstenpawlak** [reply]
I think I ran into that issue. While i was rewiring my EM24 I unplugged the USB-RS485 adapter. After plugging it in it was not recognised anymore. I only have OS v2.73 but I had v2.85 installed once and also faced that issue. Than I switched back. 
I recognised that there are 2 dbus settings for my BMS but I only have one. Its that maybe the problem?
Settings/Devices/serialbattery_ttyUSB0/ClassAndVrmInstance                                                                                                      battery:2
Settings/Devices/serialbattery_ttyUSB1/ClassAndVrmInstance                                                                                                      battery:1
The EM24 is connected to ttyUSB0.
In VRM I see data for the grid, but its alway fixed to 70W.
Is there a way to manually fix that?

**Reply by merlep** [reply]
I'm currently on V2.66 and have a ET112 measuring PV power from grid Tie inverter , 3 JKBMS's and one BMV702 - Now I want to add the 3 JK's to the Venus - only to monitor cell voltages - ( their SOC's are not accurate even after calibration ) and my BMV I have calibrated within 0.1 % SOC . So do you suggest I remain at v2.66 /2.73 ?  And how do I ensure the BMV is the primary source for SOC for the system?

**Reply by Louisvdw** [reply]
You need to manually pick the BMV as the Battery Monitor.

In the v0.12 version of the driver all the niggles for the newer VenusOS (v2.80+) have been solved. I suggest you use the latest.

**Comment by leviathanprojects** [comment]
If you have a smartshunt you can use this as a DC meter.  With latest versions of the Cerbo GX if you are using the DVCC feature it will then take the DC loads into consideration. For example if you have a charge current limit (CCL) set to 30amps and the DC meter is showing consumers of 15amps the DVCC will allow charging devices to supply 45amps. (30a for battery and 15a for the dc loads) 

This is a great feature as it means you can use DVCC as well as things like dump loads and still charge you battery at the desired rates.

**Comment by hd-schardt** [comment]
HI,
exactly this configuration I would like to implement, but so far without success. My Smartshunt is located directly behind the JK BMS in the negative branch and is configured as DC load measurement (inverter). In Venus both DC load and DVCC are enabled, but the correction of the charge power of the MPPT does not work. Purely mathematically, the correct DC load is displayed in the Venus, but without influence. Is it possibly necessary to hang the Smartshunt directly in the circuit of the DC loads, in order to capture only the pure load current, which goes to the inverter. In my previous circuit all currents from and to the battery are measured.

Greetings Heinz-Dieter
